corgi/core/models.py

```diff
@@ -940,23 +940,31 @@ def latest_components_by_streams(
         include: bool = True,
     ) -> "ComponentQuerySet":
         """Return only root components from latest builds for each product stream."""
-        product_stream_uuids = set(
+        product_stream_uuids = (
             self.using("read_only")
             .root_components()
             .exclude(productstreams__isnull=True)
             .values_list("productstreams__uuid", flat=True)
+            # Clear ordering inherited from parent Queryset, if any
+            # So .distinct() works properly and doesn't have duplicates
+            .order_by()
             .distinct()
+            .iterator()
```

Comment:
why would we use an iterator for a small flat list ? set is faster and I doubt the memory savings is anything but very small (we only have 300+ potential uuids)